zap/src/irgen/des.rs

```diff
@@ -37,13 +82,35 @@ impl Des<'_> {
 		}
 	}
 
+	fn readvariant_storage(&mut self, storage: &VariantStorageKind) -> Expr {
+		match storage {
+			VariantStorageKind::Full(numty) => self.readnumty(*numty),
+			VariantStorageKind::Bitpack(variants) => {
+				let (variant_name, variant_expr) = self.add_occurrence("variant");
+				self.push_local(variant_name.clone(), None);
+				for (i, (bits, var)) in variants.iter().enumerate() {
+					let cond = self.check_bitfield(var.clone(), *bits);
+
+					self.push_stmt(if i == 0 { Stmt::If(cond) } else { Stmt::ElseIf(cond) });
+					self.push_assign(Var::Name(variant_name.clone()), (i as f64).into());
```

Comment:
This sets `variant_name` to `i`, which doesn't feel like what you'd want to do as `i` gets overridden in the `for` loop?